### PR TITLE
Fix `best_base` to select proper base class

### PR DESCRIPTION
**Pull request by Snowapril**
This revision implements `itemsize` slots and fix `best_base` implementation to select proper base class when generating class.

### CODE REVIEW

File: Lib/test/test_types.py [hunk 1]
```diff
@@ -392,8 +392,6 @@ def test(i, format_spec, result):
         test(123456, "1=20", '11111111111111123456')
         test(123456, "*=20", '**************123456')
 
-    @unittest.expectedFailure
-    @run_with_locale('LC_NUMERIC', 'en_US.UTF8')
```
Comment (youknowone) [inline Lib/test/test_types.py:396]: This line is not expected to be removed.

File: vm/src/builtins/int.rs [hunk 2]
```diff
@@ -326,7 +326,8 @@ impl PyInt {
 
 #[pyclass(
     flags(BASETYPE),
-    with(PyRef, Comparable, Hashable, Constructor, AsNumber, Representable)
+    with(PyRef, Comparable, Hashable, Constructor, AsNumber, Representable),
+    itemsize(BigInt)
```
Comment (youknowone) [inline vm/src/builtins/int.rs:330]: I don't get how this is intended to be used yet. How to handle it if it requires multiple fields?

Comment (Snowapril) [reply vm/src/builtins/int.rs:330]: It was not intended for multiple types to be declared with `itemsize`. 
In the pyclass code, if the number of types declared with itemsize is not one, it is planned to raise an error in a way that developers can understand. Unlike `basicsize`, it seems difficult to automatically infer item type for each class delcaration, so I have chosen this way.

Comment (Snowapril) [reply vm/src/builtins/int.rs]: Hmm.. I think `itemsize` implementation is better to be moved to other PRs. It seems it does not cause wrong behavior in `best_base`.

**Comment by Snowapril** [comment]
I can't reproduce `test_types::test_int__format__locale` failure in WSL ubuntu 22.04. It also seems not related to this PR.    . 😂

File: Lib/test/test_types.py [hunk 3]
```diff
@@ -421,9 +421,6 @@ def test_int__format__locale(self):
             self.assertEqual(len(format(0, lfmt)), len(format(x, lfmt)))
             self.assertEqual(len(format(0, cfmt)), len(format(x, cfmt)))
     
-    if sys.platform != "darwin":
-        test_int__format__locale = unittest.expectedFailure(test_int__format__locale)
```
Comment (youknowone) [inline Lib/test/test_types.py:425]: Reverting this will fix the failing test

Comment (Snowapril) [reply Lib/test/test_types.py]: Oh my.. That code accidentally got included in the commit. I've fixed it!

File: vm/src/builtins/type.rs [hunk 4]
```diff
@@ -1337,28 +1355,25 @@ fn best_base(bases: &[PyTypeRef], vm: &VirtualMachine) -> PyResult<PyTypeRef> {
...
+            return Err(
+                vm.new_type_error("multiple bases have instance layout conflict".to_string())
+            );
+        }
     }
 
-    // FIXME: Ok(base.unwrap()) is expected
-    Ok(bases[0].clone())
+    debug_assert!(base.is_some());
+    Ok(base.unwrap())
```
Comment (youknowone) [inline vm/src/builtins/type.rs:1376]: `unwrap` implies `panic!`. So `debug_assert!` is duplication.

```suggestion
    Ok(base.expect("best_base must exist"))
```

**Review by youknowone — APPROVED** [review]
Thank you!

**Pull request merged**